src/spdx/parser/rdf/graph_parsing_functions.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) 2023 spdx contributors
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from enum import Enum
+from typing import Any, Callable, Optional, Type
+
+from rdflib import Graph, URIRef
+from rdflib.exceptions import UniquenessError
+from rdflib.namespace import NamespaceManager
+from rdflib.term import Node
+
+from spdx.model.spdx_no_assertion import SpdxNoAssertion, SPDX_NO_ASSERTION_STRING
+from spdx.model.spdx_none import SpdxNone, SPDX_NONE_STRING
+from spdx.parser.error import SPDXParsingError
+from spdx.parser.logger import Logger
+from spdx.rdfschema.namespace import SPDX_NAMESPACE
+from spdx.casing_tools import camel_case_to_snake_case
+
+
+def parse_literal(logger: Logger, graph: Graph, subject: Node, predicate: Node,
+                  parsing_method: Callable = lambda x: x.strip(), default: Any = None):
+    value = get_unique_value(logger, graph, subject, predicate, default)
+    if not value:
+        return default
+    return apply_parsing_method_or_log_error(logger, value, parsing_method, default)
+
+
+def apply_parsing_method_or_log_error(logger: Logger, value: Any, parsing_method: Callable = lambda x: x.strip(),
+                                      default: Any = None):
+    try:
+        return parsing_method(value)
+    except SPDXParsingError as err:
+        logger.extend(err.get_messages())
+    except (TypeError, ValueError) as err:
+        logger.append(err.args[0])
+    return default
+
+
+def parse_literal_or_no_assertion_or_none(logger: Logger, graph: Graph, subject: Node, predicate: Node,
+                                          parsing_method: Callable = lambda x: x.strip(), default: Any = None):
```

Comment:
I don't know in which cases stripping is necessary, but there are calls of this method that simply use `str` as `parsing_method` without stripping (for example the package download location). I'm not sure if this is consistent.

Reply:
I changed the default to be `strip()` rather late in the development and all cases where the parsing_method is `str` can be converted to use the default method. By stripping the values we make sure that there are no leading and trailing whitepsaces which I think is reasonable.